internal/aws/albec2/ec2.go

```diff
@@ -297,403 +170,97 @@ func (e *EC2) GetSecurityGroups(names []*string) (sgs []*string, err error) {
+	securityGroups, err := e.describeSecurityGroupsHelper(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: []*string{&sgID},
+	})
 	if err != nil {
-		return nil, nil, err
-	}
-
-	// when no results were returned, security group doesn't exist and no need to attempt modification.
-	if len(o.SecurityGroups) < 1 {
-		return nil, nil, nil
-	}
-	groupId := o.SecurityGroups[0].GroupId
-
-	// if no currentPorts were known to the LB but the sg stil resoled, query the SG to see if any ports can be resvoled
-	if len(currentPorts) < 1 {
-		currentPorts, err = e.DescribeSGPorts(groupId)
-		if err != nil {
-			return nil, nil, err
-		}
-	}
-
-	// for each addPort, run an authorize to ensure it's added
-	for _, port := range desiredPorts {
-		ipRanges := []*ec2.IpRange{}
-		for _, cidr := range desiredCidrs {
-			if existsInOtherPortRange(port, currentPorts) && existsInOtherIngressCidrRange(cidr, currentCidrs) {
-				continue
-			}
-			ipRanges = append(ipRanges, &ec2.IpRange{
-				CidrIp:      cidr,
-				Description: aws.String(fmt.Sprintf("Allow ingress on port %v from %v.", port, aws.StringValue(cidr))),
-			})
-
-		}
-
-		if len(ipRanges) == 0 {
-			continue
-		}
-
-		in := &ec2.AuthorizeSecurityGroupIngressInput{
-			GroupId: groupId,
-			IpPermissions: []*ec2.IpPermission{
-				{
-					ToPort:     aws.Int64(port),
-					FromPort:   aws.Int64(port),
-					IpProtocol: aws.String("tcp"),
-					IpRanges:   ipRanges,
-				},
-			},
-		}
-		_, err := e.AuthorizeSecurityGroupIngress(in)
-		if err != nil {
-			return nil, nil, err
-		}
+		return nil, err
 	}
-
-	// for each currentPort, run a revoke to ensure it can be removed
-	for _, port := range currentPorts {
-		ipRanges := []*ec2.IpRange{}
-		for _, cidr := range currentCidrs {
-			if existsInOtherPortRange(port, desiredPorts) && existsInOtherIngressCidrRange(cidr, desiredCidrs) {
-				continue
-			}
-			ipRanges = append(ipRanges, &ec2.IpRange{
-				CidrIp:      cidr,
-				Description: aws.String(fmt.Sprintf("Allow ingress on port %v from %v.", port, aws.StringValue(cidr))),
-			})
-		}
-
-		if len(ipRanges) == 0 {
-			continue
-		}
-
-		in := &ec2.RevokeSecurityGroupIngressInput{
-			GroupId: groupId,
-			IpPermissions: []*ec2.IpPermission{
-				{
-					ToPort:     aws.Int64(port),
-					FromPort:   aws.Int64(port),
-					IpProtocol: aws.String("tcp"),
-					IpRanges:   ipRanges,
-				},
-			},
-		}
-		_, err := e.RevokeSecurityGroupIngress(in)
-		if err != nil {
-			return nil, nil, err
-		}
+	if len(securityGroups) == 0 {
+		return nil, nil
 	}
+	return securityGroups[0], nil
+}
 
-	// attempt to resolve instance sg
-	instanceSGName := fmt.Sprintf("%s-%s", instSpecifierTag, *sgName)
-	inInstance := &ec2.DescribeSecurityGroupsInput{
+// GetSecurityGroupByName retrives securityGroup by vpcID and securityGroupName(SecurityGroup names within vpc are unique)
+func (e *EC2) GetSecurityGroupByName(vpcID string, sgName string) (*ec2.SecurityGroup, error) {
```

Comment:
Lets change `sgName` to `groupName` since its a `group-name` filter

Reply:
Also, vpcid is basically fixed for the controller, so we could resolve the vpcid in this func and save passing it in as a parameter everywhere its called

Reply:
vpc was kept as it is, since it's should be ec2 API's argument. maybe there will be multiple-vpc k8s cluster in the future(though a lot of things will break).
For now, i think it's better to resolve VPC during controller startup and saved as an configuration